atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/config/AutoScalingTargetedSweepRuntimeConfig.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue.config;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.time.Duration;
+import org.immutables.value.Value;
+
+@JsonDeserialize(as = ImmutableAutoScalingTargetedSweepRuntimeConfig.class)
+@JsonSerialize(as = ImmutableAutoScalingTargetedSweepRuntimeConfig.class)
+@Value.Immutable
+public interface AutoScalingTargetedSweepRuntimeConfig {
+    @Value.Default
+    default boolean enabled() {
+        return false; // Will eventually be removed and inlined into the existing TargetedSweep config
+    }
+
+    @Value.Default
+    default int minThreads() { // Eventually there'll be max threads too
+        return 0;
```

Comment:
would suggest `-1`: someone who wants to turn this off might miss removing `enabled` and setting to `0`

Reply:
No longer relevant, since it's now in install config and there's no need for a special placeholder (as it can derive the default from the existing config)